clients/cli/src/utils/experiment.rs

```diff
@@ -0,0 +1,20 @@
+pub struct Experiment{ name: &'static str, target_enrollment_per_1000: u64 }
+
+impl Experiment {
+    pub const CANCER_DIAGNOSTIC: Self = Self{name: "NEX-1", target_enrollment_per_1000: 10};
+
+    pub fn is_enrolled(&self, prover_id: &str) -> bool {
+        // Create a deterministic hash from the prover_id
+        let hash = md5::compute(format!("{}|{}", prover_id, self.name));
+        let hex_string = format!("{:x}", hash); // Convert hash to a hexadecimal string
+        let mut sum: u64 = 0;
+        // Sum each 8-character segment of the hex string
+        // It needs to be much more than 1000 to get an even distribution modulo 1000.
+        for i in (0..hex_string.len()).step_by(8) {
```

Comment:
I think this can be simpler by using native u128 methods. No need for loops or constructing it manually. Something like this:

    pub fn is_enrolled(&self, prover_id: &str) -> bool {
        // Create a deterministic hash from the prover_id
        let hash = md5::compute(format!("{}|{}", prover_id, self.name));

        // Convert the hash into a single u128 number
        let hash_number = u128::from_le_bytes(hash);

        // Return true if hash_number modulo 1000 is less than target_enrollment_per_1000
        (hash_number % 1000) < self.target_enrollment_per_1000
    }

Reply:
```
        let hash = md5::compute(format!("{}|{}", prover_id, self.name));
        let hash_number = u128::from_le_bytes(hash.try_into().expect("Hex string is too short"));
        (hash_number % 1000) < self.target_enrollment_per_1000
```

I tried running this code and I got `assertion failed: Experiment::CANCER_DIAGNOSTIC.is_enrolled("53")`

So it doesn't match web. I don't think we can use it for that reason, although I agree it looks better.